src/trading/ClaimProceeds.sol

```diff
@@ -0,0 +1,84 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+import 'ROOT/Mutex.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/extensions/MarketFeeCalculator.sol';
+import 'ROOT/legacy_reputation/SafeMath.sol';
+
+
+// AUDIT: Ensure that a malicious market can't subversively cause share tokens to be paid out incorrectly.
+/**
+ * @title ClaimProceeds
+ * @dev This allows users to claim their money from a market by exchanging their shares
+ */
+contract ClaimProceeds is Controlled {
+    using SafeMath for uint256;
+
+    function publicClaimProceeds(IMarket _market) onlyInGoodTimes public returns(bool) {
+        Mutex _mutex = Mutex(controller.lookup("Mutex"));
+        _mutex.acquire();
```

Comment:
Switch this to use the `ReentrancyGuard` `nonReentrant` modifier rather than manually managing the mutex.